src/DriftKernelAdapter.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Antidot\React;
+
+use Antidot\Application\Http\Application;
+use Drift\Console\OutputPrinter;
+use Drift\Server\Adapter\KernelAdapter;
+use Drift\Server\Context\ServerContext;
+use Drift\Server\Mime\MimeTypeChecker;
+use Psr\Container\ContainerInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use React\EventLoop\LoopInterface;
+use React\Filesystem\FilesystemInterface;
+use React\Promise\PromiseInterface;
+use function React\Promise\resolve;
+
+final class DriftKernelAdapter implements KernelAdapter
+{
+    private ServerContext $serverContext;
+    private MimeTypeChecker $mimeTypeChecker;
+    private string $rootPath;
+    private ContainerInterface $container;
+    private ReactApplication $application;
+    /** @var object|FilesystemInterface|null  */
+    private ?object $filesystem = null;
+
+    public function __construct(
+        ServerContext $serverContext,
+        MimeTypeChecker $mimeTypeChecker,
+        string $rootPath,
+        ?object $filesystem
+    ) {
+        $container = require $rootPath . '/config/container.php';
+        assert($container instanceof ContainerInterface);
+        $application = $container->get(Application::class);
+        assert($application instanceof ReactApplication);
+        (require $rootPath . '/router/middleware.php')($application, $container);
+        (require $rootPath . '/router/routes.php')($application, $container);
+        $this->container = $container;
+        $this->application = $application;
+        $this->serverContext = $serverContext;
+        $this->filesystem = $filesystem;
+        $this->mimeTypeChecker = $mimeTypeChecker;
+        $this->rootPath = $rootPath;
+    }
+
+    /** @psalm-suppress MixedReturnTypeCoercion */
+    public static function create(
+        LoopInterface $loop,
+        string $rootPath,
+        ServerContext $serverContext,
+        OutputPrinter $outputPrinter,
+        MimeTypeChecker $mimeTypeChecker,
+        $filesystem = null
```

Comment:
Here!

`?FilesystemInterface $filesystem`

Reply:
Doesn't like it too much, but this way I can pass psalm checks both in PHP 7.4 and 8